geometry/intermediate.py

```diff
@@ -0,0 +1,60 @@
+class Coordinate:
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+
+class LineSegment:
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+
+
+def do_line_segments_intersect(line_segment_1, line_segment_2):
+    """
+    Checks to see if two line segments have at least one point that they share
+    """
+    line_segment_2_start_orientation_relative_to_line_segment_1 = calculate_point_triplet_orientation(line_segment_1.start, line_segment_1.end, line_segment_2.start)
```

Comment:
Couldn't think of a longer name uh?

Reply:
What do you think about calling them `triplet_{segment_number}_{segment_number}_start|end}`. Something like: `triplet_1_2_start`, `triplet_1_2_end`, etc. ?